HW16/src/main/java/Task2/DynamicPageTest.java

```diff
@@ -0,0 +1,46 @@
+package Task2;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.Test;
+import org.testng.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import java.time.Duration;
+
+public class DynamicPageTest {
+    @Test
+    public void testDynamicContent() {
+        WebDriverManager.chromedriver();
+
+
+        WebDriver driver = new ChromeDriver();
+        driver.get("https://www.selenium.dev/selenium/web/dynamic.html");
+
+
+        WebElement button1 = driver.findElement(By.id("adder"));
+        button1.click();
+
+
+        new WebDriverWait(driver, Duration.ofSeconds(3)).until(ExpectedConditions.presenceOfElementLocated(By.id("box0")));
+
+
+        WebElement box = driver.findElement(By.id("box0"));
```

Comment:
1) no need to create new instance of WebDriverWait each time;
2) no need to find element twice;
```suggestion
       WebDriverWait wait = new WebDriverWait();
        WebElement box = wait(driver, Duration.ofSeconds(3)).until(ExpectedConditions.presenceOfElementLocated(By.id("box0")));
```